libs/shared/flagd-js-core/src/lib/feature-flag.ts

```diff
@@ -0,0 +1,202 @@
+import {FlagValue} from '@openfeature/server-sdk';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/**
+ * Flagd flag configuration structure.
+ */
+export class FeatureFlag {
+  private readonly _state: string;
+  private readonly _defaultVariant: string;
+  private readonly _variants: Map<string, FlagValue>;
+  private readonly _targetingString: string;
+
+  constructor(flag: any) {
+    this._state = flag['state'];
```

Comment:
```suggestion
    this._state = flag.state;
```

This will be safer once it's not an `any` type.